modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManager.java

```diff
@@ -177,41 +228,71 @@ private CompletableFuture<Void> createZoneReplicationNodes(CompletableFuture<Lis
 
             List<CompletableFuture<?>> partitionsStartFutures = new ArrayList<>();
 
-            for (int i = 0; i < assignments.size(); i++) {
-                int partId = i;
+            for (int partId = 0; partId < assignments.size(); partId++) {
+                Assignments zoneAssignment = assignments.get(partId);
 
-                partitionsStartFutures.add(createZonePartitionReplicationNodes(zoneId, partId, assignments.get(i)));
+                Assignment localMemberAssignment = localMemberAssignment(zoneAssignment);
+
+                partitionsStartFutures.add(createZonePartitionReplicationNodes(zoneId, partId, localMemberAssignment, zoneAssignment));
             }
 
             return allOf(partitionsStartFutures.toArray(new CompletableFuture<?>[0]));
         }));
     }
 
-    private CompletableFuture<Void> createZonePartitionReplicationNodes(int zoneId, int partId, Assignments assignments) {
-        if (!shouldStartLocally(assignments)) {
+    private CompletableFuture<Void> createZonePartitionReplicationNodes(
```

Comment:
Could you please add a javadoc to the method explaining its goal?

Reply:
added